### PR TITLE
blender: fix on darwin

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

Enable blender on darwin

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build openimageio
@GrahamcOfBorg build blender

File: pkgs/applications/misc/blender/default.nix [hunk 1]
```diff
@@ -23,22 +24,53 @@ stdenv.mkDerivation rec {
...
     ++ optional jackaudioSupport libjack2
     ++ optional cudaSupport cudatoolkit
     ++ optional colladaSupport opencollada;
 
   postPatch =
-    ''
+    if stdenv.isDarwin then ''
+      : > build_files/cmake/platform/platform_apple_xcode.cmake
+      substituteInPlace source/creator/CMakeLists.txt \
+        --replace '${"$"}{LIBDIR}/python' \
```
Comment (jtojnar) [inline pkgs/applications/misc/blender/default.nix]: Escaping `$` in `''` strings is done with `''`:

```suggestion
        --replace '''${LIBDIR}/python' \
```

File: pkgs/applications/misc/blender/default.nix [hunk 2]
```diff
@@ -1,13 +1,14 @@
 { config, stdenv, lib, fetchurl, boost, cmake, ffmpeg, gettext, glew
 , ilmbase, libXi, libX11, libXext, libXrender
 , libjpeg, libpng, libsamplerate, libsndfile
-, libtiff, libGLU, libGL, openal, opencolorio, openexr, openimageio, openjpeg_1, python3Packages
+, libtiff, libGLU, libGL, openal, opencolorio, openexr, openimageio, openjpeg, python3Packages
```
Comment (jtojnar) [inline pkgs/applications/misc/blender/default.nix]: Maybe it is also a time to upgrade to `openimageio2`.

File: pkgs/applications/misc/blender/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/applications/misc/blender/default.nix]:
```
nix-repl> ''              '''${test}'           ''
error: undefined variable 'test' at (string):1:22

nix-repl> ''              '${"$"}{test}'           ''
"'${test}'           "
```

Comment (jtojnar) [reply pkgs/applications/misc/blender/default.nix]: Sorry, did not realize ''' is also an escape sequence.

**Pull request merged**